src/session.ts

```diff
@@ -599,6 +602,32 @@ export class SessionManager implements Middleware {
                                         ? `${this.versionDetails.displayVersion} (${this.versionDetails.architecture})`
                                         : this.versionDetails.displayVersion,
                                     SessionStatus.Running);
+
+                                // If the user opted to not check for updates, then don't.
+                                if (!this.sessionSettings.promptToUpdatePowerShell) { return; }
+
+                                try {
+                                    const localVersion = semver.parse(this.versionDetails.version);
+                                    if (semver.lt(localVersion, "6.0.0")) {
+                                        // Skip prompting when using Windows PowerShell for now.
```

Comment:
What about some logic like if someone runs a version older than `5.1` to tell them that a new version would be available as well and point them at least to the docs? Since they still have the `Never` in this case as well, this would help a lot of people that just use the out of box version on their corporate machines running Windows 7/8.1 and don't even know there is a newer version available

Reply:
The out-of-support message is nice, although we'd need to have a list of out-of-support PowerShell versions to make that work.

The major version upgrade is probably complex enough that we should defer it to a later PR.